test/metabase/driver/sql/parameters/substitute_test.clj

```diff
@@ -166,7 +166,17 @@
                    (substitute query {"param" (i/map->FieldFilter
                                                {:field (Field (mt/id :venues field))
                                                 :value {:type  operator
-                                                        :value value}})})))))))))
+                                                        :value value}})})))))))
+    (testing "Throws if not enabled (#15488)"
+      (with-redefs [i/field-filter-operators-enabled? (constantly false)]
+        (is (= :invalid-parameter
+               (try
+                 (substitute ["select * from venues where " (param "param")]
+                             {"param" (i/map->FieldFilter
+                                       {:field (Field (mt/id :venues :price))
+                                        :value {:type  :number/>=
+                                                :value [3]}})})
+                 (catch Exception e (:type (ex-data e))))))))))
```

Comment:
Why not use `(is (thrown? ...))`? (Legitimate question, not a leading one)

Reply:
> There are two special assertions for testing exceptions.  The
"(is (thrown? c ...))" form tests if an exception of class c is
thrown:

> (is (thrown? ArithmeticException (/ 1 0))) 

> "(is (thrown-with-msg? c re ...))" does the same thing and also
tests that the message on the exception matches the regular
expression re:

These two forms check that a specific type of exception is thrown, and a way to check the message of the exception. I'm testing specifically that the exception includes a special marker that is used for error reporting: `metabase.query-processor.error-type/invalid-parameter`. This value is actually equal to `:invalid-parameter` i just didn't pull in the definition here. So here I am checking that a) it throws an exception and b) it contains information in the `ex-data` of that error. 

Probably a better solution would be to add something like `thrown-with-data?` but i haven't made that yet.